### PR TITLE
449 Added fill-heatmap layer type

### PR DESCRIPTION
**Pull request by roddyaj**
Added fill-heatmap layer type

### CODE REVIEW

File: src/reference/v8.json [hunk 1]
```diff
@@ -4923,6 +4958,149 @@
       "property-type": "data-constant"
     }
   },
+  "paint_fill-heatmap": {
+    "fill-heatmap-radius": {
```
Comment (lemmingapex) [inline src/reference/v8.json]: Does fill-heatmap-radius affect the results?  Does value of 30 do anything?  Looks like the doc is for points still, is this correct?

File: src/reference/v8.json [hunk 2]
```diff
@@ -4923,6 +4958,149 @@
...
+        "interpolated": true,
+        "parameters": [
+          "zoom",
+          "feature",
+          "feature-state"
+        ]
+      },
+      "property-type": "data-driven"
+    },
+    "fill-heatmap-weight": {
```
Comment (lemmingapex) [inline src/reference/v8.json]: Same question: Does this affect the results?

File: src/reference/v8.json (hunk 1 above)
Comment (roddyaj) [reply src/reference/v8.json]: I removed the unused properties, fill-heatmap-radius, fill-heatmap-weight, and fill-heatmap-intensity, and added fill-heatmap-limit-count.

File: src/reference/v8.json (hunk 2 above)
Comment (roddyaj) [reply src/reference/v8.json]: See above

**Comment by lemmingapex** [comment]
Thanks for making the changes, this will be useful.  Merge at will.

**Pull request merged**